Artem Sushko/Lesson5/Lesson5.Homework/Program.cs

```diff
@@ -0,0 +1,51 @@
+﻿//The greatest common divisor
+
+while (true)
+{
+    Console.Clear();
+    Console.Write("Enter first value: ");
+    var x = uint.Parse(Console.ReadLine());
+    Console.Write("Enter second value: ");
+    var y = uint.Parse(Console.ReadLine());
+
+    var z = 0;
+    for (int i = 1; i <= Math.Min(x, y); i++)
+    {
+        if (x % i == 0 && y % i == 0)
+        {
+            z = i;
+        }
+    }
+    Console.WriteLine($"The greatest common divisor is {z}!");
+
+    //The sum of the primes
+    int z1 = 0;
+    int res = 0;
+    Console.Write("\nEnter your value: ");
+    uint num = uint.Parse(Console.ReadLine());
+
+    for (int i = 2; i <= num; i++)
```

Comment:
There are a few optimizations that can be made to this code.

First, the inner loop can be optimized by breaking out of the loop as soon as a divisor is found. This will reduce the number of iterations for non-prime numbers.

```
for (int i = 2; i <= num; i++)
{
    bool is_prime = true;
    for (int k = 2; k < i; k++)
    {
        if (i % k == 0)
        {
            is_prime = false;
            break;
        }
    }
    if (is_prime)
    {
        res += i;
    }
}
```

Another optimization that can be made is to use the fact that all primes except 2 are odd. This means that we can skip checking even numbers as potential divisors in the inner loop.

```
for (int i = 2; i <= num; i++)
{
    if (i % 2 == 0)
    {
        continue;
    }
    bool is_prime = true;
    for (int k = 3; k < i; k += 2)
    {
        if (i % k == 0)
        {
            is_prime = false;
            break;
        }
    }
    if (is_prime)
    {
        res += i;
    }
}
```

Finally, you can use the sqrt() function to optimize the inner loop even further. The largest possible divisor of a number n is sqrt(n), so we can stop the loop as soon as we reach this point.

```
for (int i = 2; i <= num; i++)
{
    if (i % 2 == 0)
    {
        continue;
    }
    bool is_prime = true;
    int max_divisor = (int)Math.Sqrt(i);
    for (int k = 3; k <= max_divisor; k += 2)
    {
        if (i % k == 0)
        {
            is_prime = false;
            break;
        }
    }
    if (is_prime)
    {
        res += i;
    }
}
```

These optimizations should significantly reduce the running time of the code.